homeassistant/components/weather/openweathermap.py

```diff
@@ -129,10 +131,28 @@ def attribution(self):
         """Return the attribution."""
         return ATTRIBUTION
 
+    @property
+    def forecast(self):
+        """Return the forecast array."""
+        temp_unit = ('celsius' if self.hass.config.units.is_metric else
+                     'fahrenheit')
+
+        def create_dict(entry):
```

Comment:
I would just create the dict directly in the list comprehension, without defining this function. If you think that is not legible, I suggest going back to a regular for loop.